### DISCUSSION TITLE
JSON hash that is platform independent?

### DISCUSSION THREAD

**Discussion by WillemvJ** [nlohmann/json#4022] [Q&A]
I have a brief question, caused by the documentation provided here for the hash function of the library:

https://json.nlohmann.me/api/basic_json/std_hash/

In the example, it is noted that 

Output:

hash(null) = 2654435769
...
hash({"hello": "world"}) = 4469488738203676328

**Note the output is platform-dependent.**

I an confused by this remark. Does it mean that the hash of a certain json object  (e.g. one read from a file) will be different on different platforms? 

If yes, why would that be? Is there any way to ensure that hashes will be the same when compiling on different platforms, e.g. windows(MSVC) and linux (GCC)?

**Comment by nlohmann** [comment]
The library uses `std::hash` under the hood whose results are not guaranteed to be portable.

**Comment by WillemvJ** [comment]
Ok thanks, that makes sense. I need a platform independent-hash - I will try to convert the json to string and create a platform-independent hash of that string.

**Reply by nlohmann** [reply]
Feel free to create an MR!
